mllib/src/test/scala/org/apache/spark/mllib/tree/DecisionTreeSuite.scala

```diff
@@ -167,7 +167,7 @@ class DecisionTreeSuite extends SparkFunSuite with MLlibTestSparkContext {
     val rdd = sc.parallelize(arr)
     val strategy = new Strategy(Classification, Entropy, maxDepth = 3,
       numClasses = 2, maxBins = 100)
-    val metadata = DecisionTreeMetadata.buildMetadata(rdd.map(_.asML), strategy)
+    val metadata = DecisionTreeMetadata.buildMetadata(rdd.map(_.asML.toInstance), strategy)
```

Comment:
Rather than call `.toInstance` everywhere this is called, why not overload `buildMetadata` to accept an RDD of either one?

Reply:
I thought about this too but I was hesitant about it since it would modify source code only for the sake of making test code cleaner - I'll update it

Reply:
OK, this is fine as-is